bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,312 @@
+        scheduledPool.schedule(() -> tryConnectToCloud(localSalusApi), 1, MICROSECONDS);
+
+        this.devicePropertiesCache = Caffeine.newBuilder().maximumSize(10_000)
+                .expireAfterWrite(Duration.ofSeconds(config.getPropertiesRefreshInterval()))
+                .refreshAfterWrite(Duration.ofSeconds(config.getPropertiesRefreshInterval()))
+                .build(this::loadPropertiesForDevice);
+        this.scheduledFuture = scheduledPool.scheduleWithFixedDelay(this::refreshCloudDevices,
+                config.getRefreshInterval() * 2, config.getRefreshInterval(), SECONDS);
+
+        // Do NOT set state to online to prevent it to flip from online to offline
+        // check *tryConnectToCloud(SalusApi)*
+    }
+
+    private void tryConnectToCloud(SalusApi localSalusApi) {
+        try {
+            ApiResponse<SortedSet<Device>> response = localSalusApi.findDevices();
+            if (response.failed()) {
+                @Nullable
+                Error error = response.error();
+                if (error == null) {
+                    error = new Error(500, "Unknown error");
+                }
+                updateStatus(OFFLINE, COMMUNICATION_ERROR, error.code() + ": " + error.message());
+            } else {
+                // there is a connection with the cloud
+                updateStatus(ONLINE);
+            }
+        } catch (Exception ex) {
+            updateStatus(OFFLINE, COMMUNICATION_ERROR,
+                    "@text/cloud-bridge-handler.initialize.cannot-connect-to-cloud [\"" + ex.getMessage() + "\"]");
+        }
+    }
+
+    private void refreshCloudDevices() {
+        logger.debug("Refreshing devices from CloudBridgeHandler");
+
+        try {
+            if (!(thing instanceof Bridge bridge)) {
+                logger.debug("No bridge, refresh cancelled");
+                return;
+            }
+            List<Thing> things = bridge.getThings();
+            for (Thing thing : things) {
+                try {
+                    if (!thing.isEnabled()) {
+                        logger.debug("Thing {} is disabled, refresh cancelled", thing.getUID());
+                        continue;
+                    }
+
+                    @Nullable
+                    ThingHandler handler = thing.getHandler();
+                    if (handler == null) {
+                        logger.debug("No handler for thing {} refresh cancelled", thing.getUID());
+                        continue;
+                    }
+                    thing.getChannels().forEach(channel -> handler.handleCommand(channel.getUID(), REFRESH));
+                } catch (Exception ex) {
+                    logger.warn("Cannot refresh thing {} from CloudBridgeHandler", thing.getUID(), ex);
+                }
+            }
+
+            // all things were updated,
+            updateStatus(ONLINE);
+        } catch (Exception ex) {
+            logger.warn("Cannot refresh devices from CloudBridgeHandler", ex);
+        }
```

Comment:
This outer try/catch block can be removed.

Reply:
No it cannot.

From the guidelines:

> 31. Don't catch Exception unless an external method throws this exception. To handle unexpected expections catch RuntimeException. This can be helpful in repeated scheduled jobs to avoid stopping a refresh task due to temporary failure.

I will just downgrade to `RuntimeException`

Reply:
I left some comments because the exceptions that can occur are expected to occur, like timout, interrupt etc. Maybe have a peak on how other bindings handle this:
https://github.com/openhab/openhab-addons/blob/1d3c52a33d6bf50044ae61ced49766f7c7f80b89/bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/api/model/PlugwiseHAControllerRequest.java#L246-L278